src/scanner.cc

```diff
@@ -48,31 +39,29 @@ struct Scanner {
     indent_length_stack.clear();
     indent_length_stack.push_back(0);
 
-    if (length == 0) return;
-    size_t i = 0;
-    for (; i < length; ++i) {
-      indent_length_stack.push_back(buffer[i]);
+    if (length > 0) {
+      for (size_t i = 0; i < length; ++i) {
+        indent_length_stack.push_back(buffer[i]);
+      }
+      return;
     }
   }
 
-  void advance(TSLexer *lexer) {
-    lexer->advance(lexer, false);
-  }
+  void advance(TSLexer *lexer) { lexer->advance(lexer, false); }
 
-  void skip(TSLexer *lexer) {
-    lexer->advance(lexer, true);
-  }
+  void skip(TSLexer *lexer) { lexer->advance(lexer, true); }
 
   bool scan(TSLexer *lexer, const bool *valid_symbols) {
     if (lexer->lookahead == '\n') {
       if (valid_symbols[_NEWLINE]) {
         skip(lexer);
         lexer->result_symbol = _NEWLINE;
         return true;
-      } else return false;
+      } else
+        return false;
     }
 
-    if (lexer->get_column(lexer) == 0) {
+    if (lexer->lookahead && lexer->get_column(lexer) == 0) {
```

Comment:
the python scanner had a `while(lexer->lookahead)` around everything

Reply:
Interesting... removing the ``lexer->lookahead`` part like that in your PR makes a lot of tests fail.
Putting it back in leads to segfault, so it's clear, that's the culprit. 
Just opening a random vue with pug shows that the highlighting works at least, but the tests somehow show differences which I have to investigate.